scripts/interchain_security_testnet.sh

```diff
@@ -0,0 +1,235 @@
+
+oraid start --home $PROV_NODE_DIR \
+        --rpc.laddr tcp://${MY_IP}:26658 \
+        --grpc.address ${MY_IP}:9091 \
+        --address tcp://${MY_IP}:26655 \
+        --p2p.laddr tcp://${MY_IP}:26656 \
+        --grpc-web.enable=false \
+        &> ${PROV_NODE_DIR}/logs &
+
+tail -f ${PROV_NODE_DIR}/logs
+
+oraid q staking validators --home $PROV_NODE_DIR
+
+tee ${PROV_NODE_DIR}/consumer-proposal.json<<EOF
+{
+    "title": "Propose the addition of a new chain",
+    "description": "Gonna be a great chain",
+    "chain_id": "consumer",
+    "initial_height": {
+        "revision_height": 1
+    },
+    "genesis_hash": "Z2VuX2hhc2g=",
+    "binary_hash": "YmluX2hhc2g=",
+    "spawn_time": "2023-03-17T18:03:04.292959+07:00",
+    "consumer_redistribution_fraction": "0.75",
+    "blocks_per_distribution_transmission": 1000,
+    "historical_entries": 10000,
+    "ccv_timeout_period": 2419200000000000,
+    "transfer_timeout_period": 3600000000000,
+    "unbonding_period": 1728000000000000,
+    "deposit": "10000001orai"
+}
+EOF
+
+
+
+#create proposal
+oraid tx gov submit-proposal \
+       consumer-addition ${PROV_NODE_DIR}/consumer-proposal.json \
+       --keyring-backend test \
+       --chain-id $PROV_CHAIN_ID \
+       --from $PROV_KEY \
+       --home $PROV_NODE_DIR \
+       -b block
+
+#vote yes
+oraid tx gov vote 1  yes --from $PROV_KEY \
+       --keyring-backend test --chain-id $PROV_CHAIN_ID --home $PROV_NODE_DIR -b block
+
+#Verify that the proposal status is now `PROPOSAL_STATUS_PASSED`
+oraid q gov proposal 1 --home $PROV_NODE_DIR
+
+# Consumer chain
+CONS_NODE_DIR=~/consumer-coordinator
+rm -rf $CONS_NODE_DIR
+pkill -f interchain-security-cd
+
+mkdir -p $CONS_NODE_DIR
+CONS_NODE_MONIKER=coordinator
+CONS_CHAIN_ID=consumer
+
+interchain-security-cd init $CONS_NODE_MONIKER --chain-id $CONS_CHAIN_ID --home $CONS_NODE_DIR
+
+CONS_KEY=consumer-key
+interchain-security-cd keys add $CONS_KEY --home $CONS_NODE_DIR \
+    --keyring-backend test --output json > ${CONS_NODE_DIR}/${CONS_KEY}.json 2>&1
+
+#Get local account address
+CONS_ACCOUNT_ADDR=$(jq -r .address ${CONS_NODE_DIR}/${CONS_KEY}.json)
+
+#Add account address to genesis
+interchain-security-cd add-genesis-account $CONS_ACCOUNT_ADDR 1000000000orai \
+    --keyring-backend test --home $CONS_NODE_DIR
+
+oraid query provider consumer-genesis $CONS_CHAIN_ID --home $PROV_NODE_DIR -o json > ccvconsumer_genesis.json
+
+jq -s '.[0].app_state.ccvconsumer = .[1] | .[0]' ${CONS_NODE_DIR}/config/genesis.json ccvconsumer_genesis.json > \
+      ${CONS_NODE_DIR}/edited_genesis.json 
+
+mv ${CONS_NODE_DIR}/edited_genesis.json ${CONS_NODE_DIR}/config/genesis.json &&
+    rm ccvconsumer_genesis.json
+
+echo '{"height": "0","round": 0,"step": 0}' > ${CONS_NODE_DIR}/data/priv_validator_state.json  
+cp ${PROV_NODE_DIR}/config/priv_validator_key.json ${CONS_NODE_DIR}/config/priv_validator_key.json  
+cp ${PROV_NODE_DIR}/config/node_key.json ${CONS_NODE_DIR}/config/node_key.json
```

Comment:
Why do we have to use the exact validator & node key from the provider chain?

Reply:
During the consumer chain initialization, its validator set is populated with the unique provider chain validator. It entails that our consumer chain node is required to run using the same validator info in order to produce blocks. Hence, we have to copy the validator node keypair files into the local consumer chain node folder.